core/src/avm2/globals/flash/net/url_loader.rs

```diff
@@ -0,0 +1,175 @@
+//! `flash.net.URLLoader` builtin/prototype
+
+use crate::avm2::activation::Activation;
+use crate::avm2::class::Class;
+use crate::avm2::method::{Method, NativeMethodImpl};
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::TObject;
+use crate::avm2::value::Value;
+use crate::avm2::{Error, Object};
+use crate::backend::navigator::RequestOptions;
+use crate::loader::DataFormat;
+use crate::string::AvmString;
+use gc_arena::{GcCell, MutationContext};
+
+/// Implements `flash.net.URLLoader`'s class constructor.
+pub fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.net.URLLoader`'s instance constructor.
+pub fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        activation.super_init(this, &[])?;
+        this.set_property(
+            &QName::new(Namespace::public(), "dataFormat").into(),
+            "text".into(),
+            activation,
+        )?;
+        this.set_property(
+            &QName::new(Namespace::public(), "data").into(),
+            Value::Undefined,
+            activation,
+        )?;
+
+        if let Some(request) = args.get(0) {
+            if request != &Value::Null {
+                load(activation, Some(this), args)?;
+            }
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn bytes_loaded<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    // For now, just use `bytes_total`. The `bytesLoaded` value
+    // should really update as the download progresses, instead
+    // of jumping at completion from 0 to the total length
+    log::warn!("URLLoader.bytesLoaded - not yet implemented");
+    bytes_total(activation, this, args)
+}
+
+pub fn bytes_total<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        let data =
+            this.get_property(&QName::new(Namespace::public(), "data").into(), activation)?;
```

Comment:
FYI we have `QName::dynamic_name` to skip the `Namespace::public()` part (...we probably should rename it to `QName::public`)